packages/snaps-controllers/src/snaps/registry/json.ts

```diff
@@ -0,0 +1,103 @@
+import { SnapsRegistryDatabase } from '@metamask/snaps-registry';
+import { SnapId } from '@metamask/snaps-utils';
+import { satisfiesVersionRange } from '@metamask/utils';
+
+import {
+  SnapRegistry,
+  SnapRegistryInfo,
+  SnapRegistryRequest,
+  SnapRegistryResult,
+  SnapRegistryStatus,
+} from './registry';
+
+// TODO: Replace with a Codefi URL
+const SNAP_REGISTRY_URL =
+  'https://cdn.jsdelivr.net/gh/MetaMask/snaps-registry@main/src/registry.json';
+
+export type JsonSnapsRegistryArgs = {
+  fetchFunction?: typeof fetch;
+  url?: string;
+  failOnUnavailableRegistry?: boolean;
+};
+
+export class JsonSnapsRegistry implements SnapRegistry {
+  #url: string;
+
+  #database: SnapsRegistryDatabase | null = null;
+
+  #fetchFunction: typeof fetch;
+
+  #failOnUnavailableRegistry: boolean;
+
+  constructor({
+    url = SNAP_REGISTRY_URL,
+    fetchFunction = globalThis.fetch.bind(globalThis),
```

Comment:
Wouldn't this break if `globalThis.fetch` is undefined, i.e., in certain Node.js versions?

Reply:
Yeah, that's why we allow for an option to be passed but default to `globalThis.fetch`. We use the same pattern elsewhere